### PR TITLE
Removing mutable default arguments in core code and models

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
This addresses all cases of mutable default arguments in the core code and model libraries. I have not touched the surrogates and dmf code, which still have some of these.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by eslickj — DISMISSED** [review]
Looks good.  I didn't review it really close, but seems to be all mutable default replacement and formatting, which looks good.

**Comment by andrewlee94** [comment]
OK, the test failures here appear to be because of a change that is coming up in Pyomo 6 but has not yet been merged into the IDAES branch of Pyomo.

File: idaes/core/util/misc.py [hunk 1]
```diff
@@ -32,23 +32,26 @@
...
+def get_solver(solver=None, options=None):
     """
     General method for getting a solver object which defaults to the standard
     IDAES solver (defined in the IDAES configuration).
 
     Args:
         solver: string name for desired solver. Default=None, use default solver
         options: dict of solver options to use, overwrites any settings
-                 provided by IDAES configuration.
+                 provided by IDAES configuration. DEfault = None.
```
Comment (ksbeattie) [inline idaes/core/util/misc.py:43]: the extra "default=None" here is either redundant or needs to be explained.

**Review by ksbeattie — APPROVED** [review]
I was worried this would look awkward but I think this change actually makes things clearer.

@lbianchi-lbl, can we get an explicit check for these "dangerous" warning put into the pylint check?

**Comment by andrewlee94** [comment]
@ksbeattie I was going to suggest we explicitly add that check. However, we need to fix the surrogate folder first.

File: idaes/core/util/misc.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/core/util/misc.py]: @ksbeattie I added an explanation here, and in all the other places where a similar statement was made. However, this means I need a review again.

**Review by eslickj — APPROVED** [review]

**Review by eslickj — APPROVED** [review]

**Review by ksbeattie — APPROVED** [review]
LGTM again

**Pull request merged**